### PR TITLE
Replace "whitelist/blacklist" with "allowed/blocked"

### PR DESCRIPTION
**Pull request by gwaldo**
ref https://github.com/graphite-project/carbon/issues/567

**STATUS**: Not yet ready to merge.  Reviews and edits welcome

### CODE REVIEW

**Comment by gwaldo** [comment]
Alright, @graphite-project/committers I would love a review.  Please make it a hearty review, especially given my relative weakness with Python, and knowledge of this particular codebase.

Everything that I found to change was discovered by gratuitous grepping along the lines of `grep -rn -i -E 'whitelist|blacklist' ./*`, adding obnoxious deprecation comments so I could ` | grep -v -i deprecat` to filter what was left.

**Comment by gwaldo** [comment]
I haven't done any work on graphite-web yet.  I don't know if I'll get to work on it tonight, but we'll see.  If everything looks good, but we want to wait until both apps' PRs are ready, I'm cool with that.

File: conf/carbon.conf.example [hunk 1]
```diff
@@ -14,12 +14,12 @@
...
 #
-#   STORAGE_DIR    = $GRAPHITE_STORAGE_DIR
-#   LOCAL_DATA_DIR = %(STORAGE_DIR)s/whisper/
-#   WHITELISTS_DIR = %(STORAGE_DIR)s/lists/
-#   CONF_DIR       = %(STORAGE_DIR)s/conf/
-#   LOG_DIR        = %(STORAGE_DIR)s/log/
-#   PID_DIR        = %(STORAGE_DIR)s/
+#   STORAGE_DIR        = $GRAPHITE_STORAGE_DIR
+#   LOCAL_DATA_DIR     = %(STORAGE_DIR)s/whisper/
+#   USE_METRIC_FILTERS = %(STORAGE_DIR)s/lists/
```
Comment (obfuscurity) [inline conf/carbon.conf.example]: This should be `FILTERS_DIR` or `METRIC_FILTERS_DIR`, no?

File: conf/carbon.conf.example [hunk 2]
```diff
@@ -176,14 +176,14 @@ WHISPER_AUTOFLUSH = False
...
-# disk reads occur. Enabling this option makes it possible to decrease useless 
+# disk reads occur. Enabling this option makes it possible to decrease useless
 # page cache memory by posix_fadvise(2) with POSIX_FADVISE_RANDOM option.
 # WHISPER_FADVISE_RANDOM = False
 
-# Set this to True to enable whitelisting and blacklisting of metrics in
-# CONF_DIR/whitelist.conf and CONF_DIR/blacklist.conf. If the whitelist is
-# missing or empty, all metrics will pass through
-# USE_WHITELIST = False
+# Set this to True to enable control over what metrics are allowed to be handled
```
Comment (obfuscurity) [inline conf/carbon.conf.example]: Can we simplify this line to: `Set this to True to enable metric filters with`?

File: conf/carbon.conf.example [hunk 3]
```diff
@@ -336,13 +336,13 @@ TIME_TO_DEFER_SENDING = 0.0001
...
 # If enabled this setting is used to timeout metric client connection if no
-# metrics have been sent in specified time in seconds 
+# metrics have been sent in specified time in seconds
 #METRIC_CLIENT_IDLE_TIMEOUT = None
 
-# Set this to True to enable whitelisting and blacklisting of metrics in
-# CONF_DIR/whitelist.conf and CONF_DIR/blacklist.conf. If the whitelist is
-# missing or empty, all metrics will pass through
-# USE_WHITELIST = False
+# Set this to True to enable control over what metrics are allowed to be handled
```
Comment (obfuscurity) [inline conf/carbon.conf.example]: Same here.

File: conf/carbon.conf.example [hunk 4]
```diff
@@ -460,10 +460,10 @@ MAX_AGGREGATION_INTERVALS = 5
 # having to first wait rule.frequency seconds.
 # WRITE_BACK_FREQUENCY = 0
 
-# Set this to True to enable whitelisting and blacklisting of metrics in
-# CONF_DIR/whitelist.conf and CONF_DIR/blacklist.conf. If the whitelist is
-# missing or empty, all metrics will pass through
-# USE_WHITELIST = False
+# Set this to True to enable control over what metrics are allowed to be handled
```
Comment (obfuscurity) [inline conf/carbon.conf.example]: And here.

File: lib/carbon/conf.py [hunk 5]
```diff
@@ -191,8 +191,8 @@ class CarbonCacheOptions(usage.Options):
         ["config", "c", None, "Use the given config file."],
         ["instance", "", "a", "Manage a specific carbon instance."],
         ["logdir", "", None, "Write logs to the given directory."],
-        ["whitelist", "", None, "List of metric patterns to allow."],
-        ["blacklist", "", None, "List of metric patterns to disallow."],
+        ["allowed_metrics", "", None, "List of metric patterns to allow."],
+        ["blocked_metrics", "", None, "List of metric patterns to disallow."],
```
Comment (obfuscurity) [inline lib/carbon/conf.py]: s/disallow/block/?

**Comment by obfuscurity** [comment]
It looks like this isn't completely backwards compatible yet afa the parser and settings code. We need to be additive with the new stuff but leave the old stuff in place (not in examples/docs/confs, but in the aforementioned parser/settings). Make sense?

**Comment by gwaldo** [comment]
Thank you, @obfuscurity.  I haven't had time to look at this since Monday, but I'd had a suspicion that I'd missed one of the backwards-compatibility features.  I'll see what I can do this weekend.  I need to rebase and proceed.

I've updated the status in the first comment to reflect that this isn't yet ready for merge.

**Comment by obfuscurity** [comment]
Closing in favor of #591.

**Pull request closed without merging**